backend/danswer/indexing/indexing_pipeline.py

```diff
@@ -181,12 +183,34 @@ def index_doc_batch(
     )
 
     logger.debug("Starting chunking")
+    # chunker: Chunker = get_default_chunker(embedder)
+    # chunks: list[DocAwareChunk] = []
+    # for document in updatable_docs:
+    #     chunks.extend(chunker.chunk(document=document))
+
+    # if ENABLE_MEGA_CHUNK:
+    #     mega_chunker: Chunker = get_mega_chunker(embedder)
+    #     mega_chunks: list[DocAwareChunk] = []
+    #     for document in updatable_docs:
+    #         mega_chunks.extend(mega_chunker.chunk(document=document))
+
+    #     connect_mega_chunks(chunks=chunks, mega_chunks=mega_chunks)
+
     # The embedder is needed here to get the correct tokenizer
-    chunks: list[DocAwareChunk] = [
-        chunk
-        for document in updatable_docs
-        for chunk in chunker.chunk(document=document, embedder=embedder)
-    ]
+    chunker: Chunker = get_default_chunker(embedder)
+    if ENABLE_MEGA_CHUNK:
+        mega_chunker: Chunker = get_mega_chunker(embedder)
+    chunks: list[DocAwareChunk] = []
+    for document in updatable_docs:
+        chunks_per_document = chunker.chunk(document=document)
+        chunks.extend(chunks_per_document)
+        if ENABLE_MEGA_CHUNK:
```

Comment:
if mega chunks are enabled, we rechunk the document using the mega chunker, add references to the regular size chunks to the mega chunks, and add that to the final chunks list